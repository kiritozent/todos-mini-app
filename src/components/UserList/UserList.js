import React from 'react';
import PropTypes from 'prop-types';
import { List, ConfigProvider, Button } from 'antd';
import Spinner from '../Spinner/Spinner';
import './UserList.css';

const UserList = props => {
  const { dataSource, renderItem, getUsers, getTodos, getUsersRequest } = props;

  function renderErrorEmpty() {
    return (
      <div className="ErrorContainer">
        <Button
          type="primary"
          onClick={getUsersRequest}
          loading={getUsersRequest.fetching}
        >
          Retry
        </Button>
        <p>Data Not Found</p>
      </div>
    );
  }

  return (
    <ConfigProvider renderEmpty={getUsers.error && renderErrorEmpty}>
      <List
        dataSource={dataSource}
        loading={{
          indicator: getUsers.fetching ? <Spinner size="large" /> : <div />,
          spinning: getUsers.fetching || getTodos.fetching,
        }}
        renderItem={renderItem}
      />
    </ConfigProvider>
  );
};

UserList.propTypes = {
  dataSource: PropTypes.array.isRequired,
  renderItem: PropTypes.func.isRequired,
  getUsers: PropTypes.object.isRequired,
  getUsersRequest: PropTypes.func.isRequired,
};

export default UserList;
