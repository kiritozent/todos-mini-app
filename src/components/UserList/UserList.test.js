import React from 'react';
import { shallow } from 'enzyme';
import UserList from './UserList';

describe('UserList', () => {
  it('render without crashing passed', () => {
    const testCase = shallow(<UserList />);
    expect(testCase).toMatchSnapshot();
  });
});
