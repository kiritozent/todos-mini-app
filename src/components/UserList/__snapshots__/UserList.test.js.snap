// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList render without crashing passed 1`] = `
<ConfigProvider
  renderEmpty={false}
>
  <List
    bordered={false}
    dataSource={Array []}
    loading={
      Object {
        "indicator": <div />,
        "spinning": false,
      }
    }
    pagination={false}
    renderItem={[Function]}
    split={true}
  />
</ConfigProvider>
`;
